presto-main/src/main/java/com/facebook/presto/sql/planner/PlanOptimizers.java

```diff
@@ -444,6 +445,17 @@ public PlanOptimizers(
                                 new RemoveRedundantIdentityProjections())),
                 new RewriteIfOverAggregation(metadata.getFunctionAndTypeManager()));
 
+        builder.add(
+                new IterativeOptimizer(
+                        ruleStats,
+                        statsCalculator,
+                        estimatedExchangesCostCalculator,
+                        ImmutableSet.of(
+                                new PruneRedundantProjectionAssignments(),
+                                new InlineProjections(metadata.getFunctionAndTypeManager()),
+                                new RemoveRedundantIdentityProjections())),
```

Comment:
same again... do we really need to run this again? Or we can just merge `RemoveRedundantDistinctAggregation` into the above rule?

Always keep in mind that running optimizer is costly

Reply:
Moved this optimizer rule and get rid of these additional projection rules.